This is a temporary fix for ethernet with kernels 5.4.7+

It reverts the following change: https://patchwork.ozlabs.org/patch/1213121/
which disabled mdio init for most of the boards except NanoPi M4(V2)
or NanoPC T4 which have proper device tree definition for mdio/phy.

The proper fix will be to add phy device tree node for boards that miss
it.

---
diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
index 170c3a052b14..1f230bd854c4 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
@@ -320,7 +320,7 @@ out:
 static int stmmac_dt_phy(struct plat_stmmacenet_data *plat,
 			 struct device_node *np, struct device *dev)
 {
-	bool mdio = false;
+	bool mdio = true;
 	static const struct of_device_id need_mdio_ids[] = {
 		{ .compatible = "snps,dwc-qos-ethernet-4.10" },
 		{},
